src/firefly/js/visualize/reducer/HandlePlotChange.js

```diff
@@ -251,6 +256,15 @@ function makeNewPrimePlot(state,action) {
     return clone(state,{plotViewAry:clonePvAryWithPv(state,pv)});
 }
 
+function changeGroupLocking(state,action) {
+    var {plotGroupAry}= state;
+    var {plotId,plotGroupId,groupLocked}=  action.payload;
+    const pgIdx= getPlotGroupIdxById(state,plotGroupId);
+
+    if (pgIdx < 0) return state;
+    return updateSet(state, ['plotGroupAry',pgIdx,'lockRelated'], groupLocked);
+
```

Comment:
This looks good but you are passing too much from the dispatch. All you should need is `plotId` and `groupLocked`.
The following can be removed from the dispatch call: `plotGroupId`, `plotGroupAry`, and `plotGroup`

- `plotGroupAry` is correctly coming from the store
- `plotGroup` is not being used.
- `plotGroupId` can come from the plotView

```
const pv= getPlotViewById(state,plotId);
const {plotGroupId} = pv;
```